### ISSUE TITLE
Missing Required Key

### ISSUE THREAD

**Issue by IanMitchell** [thoughtbot/administrate#378]
Generated the admin controllers and dashboards, and made some minimal changes - getting this error when I try to open the default path, though.

```
No route matches {:action=>"show", :controller=>"admin/staff"} missing required keys: [:id]
```

This originates from the `_sidebar` file. Would this be a configuration error on my part? 

Of note, I do have the following line in my `inflections.rb` file:

```
ActiveSupport::Inflector.inflections(:en) do |inflect|
  inflect.irregular 'staff', 'staff'
end
```

Would this perhaps be causing some disconnect?

**Comment by IanMitchell** [comment]
Some updates - visiting a specific user, such as http://localhost:3000/admin/users/1 works; it appears to be only on the User index action.

Staff belongs_to user, which has_many staff.

**Comment by sikachu** [comment]
I tried to replicate this and created a simple app: https://github.com/sikachu/route-inflection

One thing I notice right away is that `rake routes` gives a different path name when `show` and `index` action both have the same name:

```
$  rake routes
           Prefix Verb   URI Pattern                     Controller#Action
admin_staff_index GET    /admin/staff(.:format)          admin/staff#index
                  POST   /admin/staff(.:format)          admin/staff#create
  new_admin_staff GET    /admin/staff/new(.:format)      admin/staff#new
 edit_admin_staff GET    /admin/staff/:id/edit(.:format) admin/staff#edit
      admin_staff GET    /admin/staff/:id(.:format)      admin/staff#show
                  PATCH  /admin/staff/:id(.:format)      admin/staff#update
                  PUT    /admin/staff/:id(.:format)      admin/staff#update
                  DELETE /admin/staff/:id(.:format)      admin/staff#destroy
```

Looking at the [`_sidebar`](https://github.com/thoughtbot/administrate/blob/4eede3b15df668e3913e7edd3a018eb97dc28aa8/app/views/administrate/application/_sidebar.html.erb#L15) generator, we seems to be using the right syntax for creating the route. I think we might want to look into how routes is defined in the app, as I tested these route generation in the test app and it works fine:

```
$ rails c
> app.url_for([:admin, Staff])
=> "http://www.example.com/admin/staff"
> app.url_for([:admin, staff])
=> "http://www.example.com/admin/staff/1"
```

Do you mind running that code and post the result on what you getting back?

**Comment by IanMitchell** [comment]
Sure thing!

```
Loading development environment (Rails 4.2.4)
irb(main):001:0>  app.url_for([:admin, Staff])
=> "http://www.example.com/admin/staff"
irb(main):002:0> app.url_for([:admin, staff])
NameError: undefined local variable or method `staff' for main:Object
	from (irb):2
	from /Users/ianm/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.2.4/lib/rails/commands/console.rb:110:in `start'
	from /Users/ianm/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.2.4/lib/rails/commands/console.rb:9:in `start'
	from /Users/ianm/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.2.4/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /Users/ianm/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.2.4/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/ianm/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/railties-4.2.4/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:4:in `require'
	from bin/rails:4:in `<main>'
irb(main):003:0> 
```

**Comment by tmecklem** [comment]
I submitted a PR for this: https://github.com/thoughtbot/administrate/pull/414

**Comment by thecalliopecrashed** [comment]
This was fixed up by #934, I believe.
Seems like it burned a lot of people, so thank you for tracking it down, @tmecklem.
